### ISSUE TITLE
Simplify `() => { ... }` to `a => { ... }`

### ISSUE THREAD

**Issue by WofWca** [terser/terser#870]
**Bug report or Feature request?**
Feature request

**Version (complete output of `terser -V` or specific git commit)**
`v5.3.8`

**Complete CLI command or `minify()` options used**
```js
{
  compress: {
    unsafe: true,
  },
  mangle: {},
  output: {},
  parse: {},
  rename: {},
}
```

**`terser` input**
```js
const fn = () => { something(); };
```

**`terser` output or error**
```js
const fn=()=>{something()};
```

**Expected result**
```js
const fn=a=>{something()};
```

In other words, for arrow functions with no parameters, specify a dummy (unused) parameter instead of writing empty parentheses.

Not a huge JS expert.
* The main concern is whether arguments can always be skipped.
* Another concern is about the `arguments` built-in object, but they don't apply to arrow functions.
* One more concern is that in some cases (e.g. when there's a crazy amount of nested arrow functions) it may actually increase the output size because it consumes available variable names.

**Comment by fabiosantoscode** [comment]
Hello!

I think this is a valid optimization to do, but since it only saves one byte, it's not worth it in my opinion.

**Comment by WofWca** [comment]
One byte per occurrence yes, however, such functions are very common.
I gave it a shot, out of interest mostly. Turned out not super good, but not super bad either: #874.

Tasks that I came up with while I was busy with this:
* What to do with numerous existing tests that use such functions, they're failing now. Add a `minify` option for this minification and turn it off by default?
* It transforms this
```js
a = {
  b: () => { doStuff(); },
}
```
into this
```js
a = {
  b(_) { doStuff(); }
}
```
Consider this solved by adding more `compress.passes`?
* When such options are provided that the dummy parameter name is not mangled, it can output its name as `a$0` (see `create_symbol` function), which makes the output less optimal.
* `return() =>` should be better than `return a =>`. Same for `export default` and `throw` and maybe something else.
* Would be cool if the dummy parameter name could also get shadowed by variables declared in the function scope, to save some namespace space.
